### ISSUE TITLE
DMTF Redfish integrate

### ISSUE THREAD

**Issue by shan100github** [tinkerbell/roadmap#26]
[DMTF Redfish](https://www.dmtf.org/standards/redfish)
Redfish being created as DMTF’s Redfish® is a standard designed to deliver simple and secure management for converged, hybrid IT and the Software Defined Data Center (SDDC). Both human readable and machine capable, Redfish leverages common Internet and web services standards to expose information directly to the modern tool chain.

Tinkerbell might be the best place to provide an integration with redfish to manage hardware.
Please share your thoughts and opinion.

**Comment by chrisdoherty4** [comment]
What kinds of operations are you thinking we'd need to utilize; are they additional to what Tinkerbell already does (power and boot device manipulation)? 

Tinkerbell can already use Redfish for its BMC operations. The components that integrate with baseboard management controllers leverage a library called [`bmclib`](https://github.com/bmc-toolbox/bmclib). It provides Redfish integration for us and we contribute where we can to keep it up-to-date.

**Comment by shan100github** [comment]
yes in addition to current  (power and boot device manipulation), if possible it would be good to use it to perform monitoring, alerts, and notifications of underlying hardware.

I am not sure does it comes under tinkerbell's goal.

**Comment by chrisdoherty4** [comment]
> monitoring, alerts, and notifications of underlying hardware.

What properties do you envisage monitoring and alerting on? Is there a Redfish API that lets you 'watch' hardware for events?

> I am not sure does it comes under Tinkerbell's goal.

A monitoring capability would certainly be a scope expansion to Tinkerbell. It wouldn't fit within the current goals of the project per-se, but goals can be expanded also.

---

I support this sitting in the roadmap repository as an 'on ice' issue meaning we hold off adding it as a [project item](https://github.com/orgs/tinkerbell/projects/16?query=is%3Aopen+sort%3Aupdated-desc) due to the scope increase it would bring. 

I think its also important to manage expectations and I suspect the current maintainers won't be able to prioritize it for some time due to the backlog of existing work. Additional contributors are welcome to champion.

**Comment by shan100github** [comment]
It will be helpful to monitor the hardware status of Hardware like LAN, RAM etc, along with the support to provision OS and the environment through Tinkerbell.

Some details about monitoring and events can be referred from https://www.dmtf.org/sites/default/files/Redfish%20School%20-%20Events.pdf

**Comment by pedroalvesbatista** [comment]
@chrisdoherty4 @jacobweinstock 

Could be this the case for a plugin or add-on so people can include more features or extend Tinkerbell capabilities ?
